### PR TITLE
Add BIR spec test

### LINKED ISSUE #24774: Write a unit test for BIR spec for hello world program
Get the BIR spec working for hello world and write a unit test for the spec.

Sub task of #24771

### PR DESCRIPTION
**Pull request by Kishanthan**
## Purpose
This PR will add a validation test for BIR binary by reading and comparing it with original BIR model.

Resolves #24774

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: docs/bir-spec/src/test/resources/test-src/basic/function.bal [hunk 1]
```diff
@@ -0,0 +1,3 @@
+function testFunction() {
```
Comment (riyafa) [inline docs/bir-spec/src/test/resources/test-src/basic/function.bal:1]: We add license header for test cases as well if I am not mistaken

File: docs/bir-spec/src/test/java/org/ballerinalang/birspec/BIRSpecTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Test cases to verify BIR binary against the spec.
+ */
+public class BIRSpecTest {
+
+    @Test(description = "A simple test to verify BIR binary of a ballerina source with an empty method")
+    public void basicTest() {
```
Comment (riyafa) [inline docs/bir-spec/src/test/java/org/ballerinalang/birspec/BIRSpecTest.java:40]: There seems to be lots of work in validating a single bir? Can't we simplify this. I mean some assert function that would validate the expected and the current would be useful.

Comment (riyafa) [reply docs/bir-spec/src/test/java/org/ballerinalang/birspec/BIRSpecTest.java:40]: Then the expected and the current would be more obvious. I think they are hidden in some complex code at the present

File: docs/bir-spec/src/test/resources/test-src/basic/function.bal (hunk 1 above)
Comment (Kishanthan) [reply docs/bir-spec/src/test/resources/test-src/basic/function.bal]: ack

File: docs/bir-spec/src/main/resources/kaitai/bir.ksy [hunk 3]
```diff
@@ -185,6 +198,95 @@ types:
...
+            'type_tag_enum::type_tag_byte': byte_constant_info
+            'type_tag_enum::type_tag_float': float_constant_info
+            'type_tag_enum::type_tag_string': string_constant_info
+            'type_tag_enum::type_tag_decimal': decimal_constant_info
+            'type_tag_enum::type_tag_boolean': boolean_constant_info
+            'type_tag_enum::type_tag_nil': nil_constant_info
+            'type_tag_enum::type_tag_map': map_constant_info
+    instances:
+      type_tag:
+        value: _root.constant_pool.constant_pool_entries[constant_value_type_cp_index].cp_info.as<shape_cp_info>.shape.type_tag
```
[the remark this answers is not in the document]
Comment (Kishanthan) [reply docs/bir-spec/src/main/resources/kaitai/bir.ksy]: ack. changed.

**Review by manuranga — APPROVED** [review]

**Pull request merged**